src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java

```diff
@@ -0,0 +1,139 @@
+              "TransactionDesc": "Payment of X"
+            }
+         */
+        from("rest:POST:/buygoods")
+                .id("buy-goods-online")
+                .process(exchange -> {
+                    String body = exchange.getIn().getBody(String.class);
+                    BuyGoodsPaymentRequestDTO buyGoodsPaymentRequestDTO = objectMapper.readValue(
+                            body, BuyGoodsPaymentRequestDTO.class);
+
+                    exchange.setProperty(BUY_GOODS_REQUEST_BODY, buyGoodsPaymentRequestDTO);
+                    logger.info(body);
+                    logger.info(buyGoodsPaymentRequestDTO.toString());
+
+                })
+                .to("direct:buy-goods-base");
+
+        /*
+         * Starts the payment flow
+         *
+         * Step1: Authenticate the user by initiating [get-access-token] flow
+         * Step2: On successful [Step1], directs to [lipana-buy-goods] flow
+         */
+        from("direct:buy-goods-base")
+                .id("buy-goods-base")
+                .log(LoggingLevel.INFO, "Starting buy goods flow")
+                .to("direct:get-access-token")
+                .process(exchange -> {
+                    accessTokenStore.setExpiresOn(3599);
+                    exchange.setProperty(ACCESS_TOKEN, accessTokenStore.getAccessToken());
+
+                })
+                .log(LoggingLevel.INFO, "Got access token, moving on to API call.")
+                .to("direct:lipana-buy-goods");
+
+        /*
+         * Takes the access toke and payment request and forwards the requests to lipana API.
+         * [Password] and [TransactionType] are set in runtime and request is forwarded to lipana endpoint.
+         */
+        from("direct:lipana-buy-goods")
+                .removeHeader("*")
+                .setHeader(Exchange.HTTP_METHOD, constant("POST"))
+                .setHeader("Content-Type", constant("application/json"))
+                .setHeader("Authorization", simple("Bearer ${exchangeProperty."+ACCESS_TOKEN+"}"))
+                .setBody(exchange -> {
+                    BuyGoodsPaymentRequestDTO buyGoodsPaymentRequestDTO =
+                            (BuyGoodsPaymentRequestDTO) exchange.getProperty(BUY_GOODS_REQUEST_BODY);
+
+
+                    String password = getPassword("" + buyGoodsPaymentRequestDTO.getBusinessShortCode(),
+                            passKey,
+                            "" + buyGoodsPaymentRequestDTO.getTimestamp());
+
+                    buyGoodsPaymentRequestDTO.setPassword(password);
+                    buyGoodsPaymentRequestDTO.setTransactionType(MPESA_BUY_GOODS_TRANSACTION_TYPE);
+
+                    logger.info(buyGoodsPaymentRequestDTO.toString());
+                    logger.info(accessTokenStore.getAccessToken());
+
+                    return buyGoodsPaymentRequestDTO;
+                })
+                .marshal().json(JsonLibrary.Jackson)
+                .toD(buyGoodsBaseUrl+"?bridgeEndpoint=true");
```

Comment:
Are you getting the correct response back?

Reply:
yes